src/internal/require/require.go

```diff
@@ -419,21 +421,11 @@ func NoErrorWithinT(tb testing.TB, t time.Duration, f func() error, msgAndArgs .
 // emit an error. Unlike NoErrorWithinT if f does error, it will retry it.
 func NoErrorWithinTRetry(tb testing.TB, t time.Duration, f func() error, msgAndArgs ...interface{}) {
 	tb.Helper()
-	doneCh := make(chan struct{})
-	timeout := false
-	var err error
-	go func() {
-		for !timeout {
-			if err = f(); err == nil {
-				close(doneCh)
-				break
-			}
-		}
-	}()
-	select {
-	case <-doneCh:
-	case <-time.After(t):
-		timeout = true
+	ctx, _ := context.WithTimeout(context.Background(), t)
```

Comment:
We should still have the cancel in a defer in the case of an error before the timeout.